ConsentViewController/Classes/SPConsentManager.swift

```diff
@@ -177,45 +177,111 @@ import UIKit
         }
     }
 
+    // TODO: move this code to SPClientCoordinator
     func report(action: SPAction) {
         responsesToReceive += 1
         switch action.campaignType {
         case .ccpa:
-            spClient.postCCPAAction(authId: authId, action: action, localState: storage.localState, idfaStatus: idfaStatus) { [weak self] result in
-                self?.responsesToReceive -= 1
-                switch result {
-                case .success((let localState, let consents)):
-                    let userData = SPUserData(
-                        gdpr: self?.storage.userData.gdpr,
-                        ccpa: SPConsent(consents: consents, applies: true)
-                    )
-                    self?.storeData(
-                        localState: localState,
-                        userData: userData,
-                        propertyId: self?.propertyId
+                var pmPayload: CCPAChoiceBody.PMSaveAndExitPayload?
+                if let rCat = action.pmPayload["rejectedCategories"]?.anyValue as? [String?],
```

Comment:
pmPayload will be parsed from the rendering app into a custom `struct` instead of it being a generic `SPJson`